TFEngine.py

```diff
@@ -1344,48 +1345,84 @@ def _do_save(self):
         return False
     return True
 
-  def eval_model(self, output_file=None, output_per_seq_file=None):
+  def eval_model(self, output_file=None, output_per_seq_file=None, loss_name=None, output_per_seq_format="seq_len;score;pos_score"):
     """
     Eval the current model on the eval datasets (dev + eval, whatever is set).
     See also :func:`self.search` for performing beam search.
 
     :param str|None output_file: if given, will save the results to this file (total err/score for each dataset)
     :param str|None output_per_seq_file: if given, will save the err/score for each sequence
+    :param [str]|None loss_name=None: Needed if output_file is given.
+    :param str|None output_per_seq_format="seq_len;score;pos_score":
+                                            which properies of `loss_name` should be written to `output_per_seq_file`.
+                                            allowed_outputs = {"seq_tags", "seq_len", "score", "error", "pos_score", "pos_error"}
+                                            requested outputs must be separated by a colon ";", e.g. "seq_len;score;error"
+
     :return: nothing
     """
+
     extra_fetches = None
-    results_per_seq = {}  # seq_tag -> dict[str,float]
+    requested_outputs=[]
+
     if output_per_seq_file:
-      extra_fetches = {}
+      allowed_outputs = {"seq_tags", "seq_len", "score", "error", "pos_score", "pos_error"}
+
+      requested_outputs = output_per_seq_format.split(';')
```

Comment:
As said, do not use `split` here, and also no need for the extra `output_per_seq_format` variable. (Actually, maybe that is a better param name?)
Instead, just do:
```suggestion
      assert isinstance(output_per_seq_format, (tuple, list))
```